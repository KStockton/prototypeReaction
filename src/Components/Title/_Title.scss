@import '../../Sass/variables.scss'; 

.Title-btn-wrapper {
  @include displayFlex(space-evenly, null, null, null);
  padding-bottom: 7rem;

 button {
   width: 120px;
   height: 55px;
   margin-bottom: 57px;
   border-radius: 8px;
   border-color: ghostwhite;
   background-color: rgba(16, 23, 16, 0.509);
   color: $btn-text-color;
   font-size: 20px;
   cursor: pointer;
   &:hover {
     background-color: rgba(142, 235, 203, 0.55);
    }
  }
}
.toggler {
  position: relative;
  z-index: 2;
  width: var(--toggler-width);
  height: var(--toggler-height);
  background-color: var(--toggler-bg-color);
  border: none;
  border-radius: 5px;
  outline: none;
  cursor: pointer;
  transition: background-color 350ms;
}

.toggler--active {
  background-color: var(--toggler-active-bg-color);
}

.menu {
  position: absolute;
  top: var(--menu-ending-top);
  z-index: 1;
  box-sizing: border-box;
  width: var(--menu-width);
  padding: 0 20px;
  overflow: hidden;
  background-color: var(--menu-bg-color);
  border-radius: 5px;
}

.list {
  padding: 0;
  list-style-type: none;
}

.list-item {
  padding: 5px 0;
}

.display-enter {
  top: var(--menu-starting-top);
  width: var(--toggler-width);
  max-height: var(--toggler-height);
  color: var(--fade-from-color);
  background-color: var(--toggler-bg-color);
}

.display-enter-active {
  top: var(--menu-ending-top);
  width: var(--menu-width);
  max-height: var(--menu-max-height);
  color: var(--fade-to-color);
  background-color: var(--menu-bg-color);
  transition: all var(--timeout);
}

.display-exit {
  top: var(--menu-ending-top);
  width: var(--menu-width);
  max-height: var(--menu-max-height);
  color: var(--fade-to-color);
  background-color: var(--menu-bg-color);
}

.display-exit-active {
  top: var(--menu-starting-top);
  width: var(--toggler-width);
  max-height: var(--toggler-height);
  color: var(--fade-from-color);
  background-color: var(--toggler-bg-color);
  transition: all var(--timeout);
}

   
