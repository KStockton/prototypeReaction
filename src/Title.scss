.Title-btn-wrapper {
 display: flex;
 justify-content: space-evenly;

 button {
   width: 120px;
   height: 55px;
   margin-bottom: 57px;
   border-radius: 8px;
   border-color: ghostwhite;
   background-color: rgba(16, 23, 16, 0.509);
   color: white;
   font-size: 20px;
   &:hover {
    background-color: rgba(142, 235, 203, 0.55);
   }
 }
}