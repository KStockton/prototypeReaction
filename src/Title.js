import React, { Component } from 'react'
import './Title.scss'

export default class Title extends Component {

  handleClick = (event) => {
   
    let studyOption = event.target.value;

    switch(studyOption) {
      case "Array Prototypes":
        this.props.selectTopic(studyOption);
        break
      case "String Prototypes":
        this.props.selectTopic(studyOption);
        break
      default: this.props.selectReview()
    }
  }

  render() {
   return (
    <section className="Title-btn-wrapper bounce">
         <button value="Array Prototypes" name="arrayEvent" onClick={this.handleClick}>Array Prototypes</button>
         <button value="review" onClick={this.handleClick}> Review Cards</button>
         <button value="String Prototypes" name="stringEvent" onClick={this.handleClick}>String Prototypes</button>
    </section> 
   )
  }
}