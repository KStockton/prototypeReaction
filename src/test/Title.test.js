import React from 'react'
import Title from '../Title'
import { shallow } from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

const mock_handleClick = jest.fn();
const

describe('Title', () =>{
 let wrapper

   beforeEach(() =>{
    wrapper = shallow(<Title handleClick={mock_handleClick}/>
    )
   });
   it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call updateSearch when search form is submitted', () => {
    wrapper.find('.array').simulate('click', { target: { value: 'array' } });
    expect(mock_handleClick).toBeCalled();
    // expect(mock_handleClick).toBeCalled();
  });


});