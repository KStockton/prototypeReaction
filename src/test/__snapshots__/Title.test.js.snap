// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match the snapshot with all the data passed in 1`] = `
<section
  className="Title-btn-wrapper bounce"
>
  <button
    className="array"
    name="arrayEvent"
    onClick={[Function]}
    value="Array Prototypes"
  >
    Array Prototypes
  </button>
  <button
    onClick={[Function]}
    value="review"
  >
     Review Cards 
  </button>
  <button
    name="stringEvent"
    onClick={[Function]}
    value="String Prototypes"
  >
    String Prototypes
  </button>
</section>
`;
